worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageMappingCallable.java

```diff
@@ -173,107 +170,37 @@ private boolean filterByYear(org.hl7.fhir.dstu3.model.Patient patient) {
      * @param patient - the patient id
      * @return patientId if present, null otherwise
      */
-    private Identifiers extractPatientId(org.hl7.fhir.dstu3.model.Patient patient) {
-        List<org.hl7.fhir.dstu3.model.Identifier> identifiers = patient.getIdentifier();
-
+    private Identifiers extractPatientId(IDomainResource patient) {
         // Get patient beneficiary id
         // if not found eobs cannot be looked up so do not return a meaningful list
-        Optional<String> beneId = getBeneficiaryId(identifiers);
-        if (beneId.isEmpty()) {
+        String beneId = IdentifierUtils.getBeneId(patient);
+        if (beneId == null || beneId.isEmpty()) {
             missingBeneId += 1;
             return null;
         }
 
         // Get current mbi if present or else log not present
-        String currentMbi = getCurrentMbi(identifiers);
+        String currentMbi = IdentifierUtils.getCurrentMbi(patient);
         if (currentMbi == null) {
             missingCurrentMbi += 1;
         }
 
-        LinkedHashSet<String> historicMbis = getHistoricMbis(identifiers, currentMbi);
+        Set<String> historicMbis = IdentifierUtils.getHistoricMbi(patient);
```

Comment:
list of mbis could be null and would cause npe

Reply:
ok